micromamba/src/env.cpp

```diff
@@ -120,8 +120,15 @@ set_env_command(CLI::App* com)
                             dependencies << "=" << v.build_string;
                         dependencies << "\n";
                     }
-
-                    channels.insert(make_channel(v.url).base_url());
+                    const auto& chan_name = make_channel(v.url).name();
+                    if (chan_name == "pkgs/main")
+                    {
+                        channels.insert("defaults");
+                    }
```

Comment:
It looks like using either `pkgs/main` or `defaults` works so I would be conservative and say we do not need to change? I'm no expert here. What do you think?

In the refactorign reflection, this should go in `Channel`, eventually through a new method name like `cannonical_name` so that we don't manage edge cases everywhere.

Reply:
Yes we could leave `pkgs/main` then (less prone to bugs and avoid edge cases). And yes, I agree that we should use something like `canonical_name` like Conda does IIRC.